packages/typespec-test/test/batch_modular/generated/typespec-ts/src/api/pagingHelpers.ts

```diff
@@ -0,0 +1,233 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import {
+  Client,
+  createRestError,
+  PathUncheckedResponse,
+} from "@azure-rest/core-client";
+import { RestError } from "@azure/core-rest-pipeline";
+import {
+  ContinuablePage,
+  PageSettings,
+  PagedAsyncIterableIterator,
+} from "../models/pagingTypes.js";
+import { isUnexpected } from "../rest/index.js";
+
+/**
+ * An interface that describes how to communicate with the service.
+ */
+interface PagedResult<
+  TElement,
+  TPage = TElement[],
+  TPageSettings = PageSettings
+> {
+  /**
+   * Link to the first page of results.
+   */
+  firstPageLink?: string;
```

Comment:
@deyaaeldeen I notice the property is required in core definition but here this would be optional because we can call from getInitialResponse in this first time.

Reply:
Yeah, it is required in case the implementation will send the first call to get the first page, but if we can that first page outside the implementation, then yeah it doesn't have to be required.